File: ts/128/34.ts
function searchRange(nums: number[], target: number): number[] {
  const l = searchIndex(nums, target, true)
  if (nums[l] !== target) return [-1, -1]
  const r = searchIndex(nums, target, false)
  return [l, r]
};

function searchIndex (nums: number[], target: number, isLeft: boolean): number  {
  let l = 0, r = nums.length - 1, m = 0
  while (l <= r) {
    m = l + ((r - l) >> 1)
    if (nums[m] > target) r = m - 1
    else if (nums[m] < target) l = m + 1
    else {
      if (isLeft) r = m - 1
      else l = m + 1
    }
  }
  return isLeft ? l : r
}

File: py3/128/34.py
from typing import List

class Solution:
    def searchRange(self, nums: List[int], target: int) -> List[int]:
        def searchIndex(nums, target, isLast):
            l, r = 0, len(nums) - 1
            while l <= r:
                m = l + (r - l) // 2
                if nums[m] == target:
                    if isLast:
                        l = m + 1
                    else:
                        r = m - 1
                elif nums[m] > target:
                    r = m - 1
                else:
                    l = m + 1
            if isLast:
                return r
            return l

        first = searchIndex(nums, target, False)
        last = searchIndex(nums, target, True)
        if first <= last and nums[first] == target:
            return [first, last]
        return [-1, -1]
